battle-service/src/main/java/by/ibondarau/tictactoe/battleservice/controller/BattleController.java

```diff
@@ -13,88 +20,97 @@
-                .firstPlayerId(1)
-                .secondPlayerId(joinBattleDto.secondPlayerId())
-                .size(3)
-                .status("Started")
-                .moves(Collections.emptyList())
-                .nextMove(1)
-                .result(null)
-                .created(ZonedDateTime.now().toInstant())
-                .started(ZonedDateTime.now().toInstant())
-                .finished(null);
-        return ResponseEntity.ok(dto);
+    public ResponseEntity<BattleResponseDto> joinBattle(@PathVariable UUID battleId, @RequestBody JoinBattleDto joinBattleDto) {
+        log.debug("""
+                        BattleController.joinBattle() - started
+                        Input parameters: battleId={}, playerId={}""",
+                battleId, joinBattleDto.toString());
+
+        BattleResponseDto battleResponseDto = modelMapper.battleToBattleResponse(
+                battleService.joinBattle(battleId, joinBattleDto.getSecondPlayerId())
+        );
+        return ResponseEntity.ok(battleResponseDto);
     }
 
     @PostMapping("/{battleId}/move")
-    public ResponseEntity<BattleResponseDto> makeMove(@PathVariable Integer battleId, @RequestBody MoveDto nextMove) {
-        BattleResponseDto dto = new BattleResponseDto()
-                .id(battleId)
-                .firstPlayerId(1)
-                .secondPlayerId(2)
-                .size(3)
-                .status("Started")
-                .moves(Collections.singletonList(nextMove))
-                .nextMove(2)
-                .result(null)
-                .created(ZonedDateTime.now().toInstant())
-                .started(ZonedDateTime.now().toInstant())
-                .finished(null);
-        return ResponseEntity.ok(dto);
+    public ResponseEntity<BattleResponseDto> makeMove(@PathVariable UUID battleId, @RequestBody MoveDto nextMove) {
+        log.debug("""
+                BattleController.makeMove() - started
+                Input parameters: battleId={}, nextMove={}""", battleId, nextMove);
+
+        Battle battle = battleService.makeMove(battleId, modelMapper.moveDtoToMove(nextMove));
+        return ResponseEntity.ok(modelMapper.battleToBattleResponse(battle));
     }
 
     @GetMapping
-    public ResponseEntity<List<BattleResponseDto>> searchBattles(
-            @RequestParam("active") Boolean active,
-            @RequestParam("pageSize") Integer pageSize,
-            @RequestParam("pageNum") Integer pageNum) {
-        return ResponseEntity.ok(new ArrayList<>());
+    public ResponseEntity<List<BattleResponseDto>> findBattles(
+            @RequestParam("status") List<BattleStatus> statuses,
+            @RequestParam("pageNum") Integer pageNum,
+            @RequestParam("pageSize") Integer pageSize) {
+        log.debug("""
+                BattleController.findBattles - Started
+                Input parameters: statuses={}, pageNum={}, pageSize={}""", statuses, pageNum, pageSize);
+        validateStatuses(statuses);
```

Comment:
you can implement spring validator and validate request param via `@Valid` annotation

Reply:
I removed custom validation here because I found an easy way to do everything I need with built-in tools just replacing list with a set, and it works quite well. Although, I tried to implement custom validator here, I spent several hours trying to make already written validator work and I didn't succeed. I found a lot of contradicting guides on the Internet and none of them eventually worked for me.